### PR TITLE
 Drop xmlparsedata and replace it with getParseData

### PR DESCRIPTION
**Pull request by Ellpeck**

### CODE REVIEW

**Comment by Ellpeck** [comment]
Some notes from @EagleoutIce:
- We want to rip out the old `xmlparsedata` parsing completely as part of this PR, this is meant to be a full replacement
- We also want to stop using xml altogether and parse the table csv straight into a proper tree

**Comment by Ellpeck** [comment]
oh fucking hell

**Comment by Ellpeck** [comment]
This is pretty much review-ready, except for the type safety of `XmlBasedJson` which is quite a lot of work

File: test/functionality/_helper/shell.ts [hunk 1]
```diff
@@ -39,35 +39,14 @@ export const testWithShell = (msg: string, fn: (shell: RShell, test: Mocha.Conte
 	})
 }
 
-function installWarning(pkg: string) {
```
Comment (EagleoutIce) [inline test/functionality/_helper/shell.ts:42]: teeechnisch gesehen scheitern jetzt die zugehörigen statistics tests oder? (in der Hoffnung das es sie gibt :P)

**Review by EagleoutIce — APPROVED** [review]

**Comment by EagleoutIce** [comment]
_This pull request is included in v1.4.2 (see [Release v1.4.2 (Dropping xmlparsedata, Benchmark Re-Runs, and Repl Fixes)](https://github.com/Code-Inspect/flowr/releases/tag/v1.4.2))._

**Pull request merged**